wfe2/wfe.go

```diff
@@ -2119,6 +2119,83 @@ func (wfe *WebFrontEndImpl) refundNewOrderLimits(ctx context.Context, transactio
 	}
 }
 
+// orderMatchesReplacement checks if the order matches the provided certificate
+// as identified by the provided ARI CertID. This function ensures that:
+//   - the certificate being replaced exists,
+//   - the requesting account owns that certificate, and
+//   - the names in this new order match the names in the certificate
+//     being replaced.
+func (wfe *WebFrontEndImpl) orderMatchesReplacement(ctx context.Context, acct *core.Registration, names []string, serial string) (*probs.ProblemDetails, error) {
+	oldCert, err := wfe.sa.GetCertificate(ctx, &sapb.Serial{Serial: serial})
+	if err != nil {
+		if errors.Is(err, berrors.NotFound) {
+			return probs.NotFound("Existing certificate could not found"), err
+		}
+		return probs.ServerInternal("Failed to retrieve existing certificate"), err
+	}
+
+	if oldCert.RegistrationID != acct.ID {
+		return probs.Unauthorized("Requester account did request the certificate being replaced by this order"), nil
+	}
+	parsedCert, err := x509.ParseCertificate(oldCert.Der)
+	if err != nil {
+		return probs.ServerInternal("Error parsing certificate replaced by this order"), err
+	}
+
+	// While draft-ietf-acme-ari-02 calls for the new order to "share a
+	// preponderance of identifiers" with the certificate being replaced, our
+	// implementation will be more lenient, requiring just one.
```

Comment:
Why just one? Since @aarongable is writing the proposed standard, shouldn't we follow it as written or should the proposed standard change? To me it's, "Here's a standard and an immediate divergence which will be documented _somewhere_."

If you decide to keep it as a single match, change `var namesMatch bool` to `var nameMatch bool`.

Reply:
We discussed this as a team and reached the conclusion documented here. The crux of the argument for a single identifier was, "the requester is stating an intent here, regardless of how many identifiers are changing they are stating plainly that this certificate replaces a predecessor." We check that they've got at least one matching identifier to be sure that this request is at least _somewhat_ similar to the one we're replacing.

Reply:
And I am going to update the draft in the next day or two anyway, and can simply remove the word "preponderance" to make this less of a departure.